tests/manage/test_pvc_deletion_during_io.py

```diff
@@ -4,62 +4,51 @@
 from ocs_ci.ocs import exceptions, constants
 from ocs_ci.ocs.resources import pod
 from ocs_ci.framework.testlib import ManageTest, tier1
-from tests.fixtures import (
-    create_rbd_storageclass, create_rbd_pod, create_pvc, create_ceph_block_pool,
-    create_rbd_secret
-)
 
 logger = logging.getLogger(__name__)
 
 
-@pytest.mark.usefixtures(
-    create_rbd_secret.__name__,
-    create_ceph_block_pool.__name__,
-    create_rbd_storageclass.__name__,
-    create_pvc.__name__,
-    create_rbd_pod.__name__,
-)
 @pytest.mark.polarion_id("OCS-371")
 class TestDeletePVCWhileRunningIO(ManageTest):
     """
     Delete PVC while IO is in progress
     """
 
     @tier1
-    def test_run_io_and_delete_pvc(self):
+    def test_run_io_and_delete_pvc(self, rbd_pod):
         """
         Delete PVC while IO is in progress
         """
-        thread = pod.run_io_in_bg(self.pod_obj, expect_to_fail=True)
-        self.pvc_obj.delete(wait=False)
+        thread = pod.run_io_in_bg(rbd_pod, expect_to_fail=True)
+        rbd_pod.pvc.delete(wait=False)
 
         # This is a workaround for bug 1715627 (replaces wait_for_resource)
-        pvc_out = self.pvc_obj.get(out_yaml_format=False)
+        pvc_out = rbd_pod.pvc.get(out_yaml_format=False)
         assert constants.STATUS_TERMINATING in pvc_out, (
-            f"PVC {self.pvc_obj.name} "
+            f"PVC {rbd_pod.pvc.name} "
             f"failed to reach status {constants.STATUS_TERMINATING}"
         )
 
         thread.join(timeout=15)
 
-        self.pod_obj.delete()
+        rbd_pod.delete()
```

Comment:
Will we hit issues in `rbd_pod`'s finalizer if we are calling `.delete()` here in the test?

Reply:
@clacroix12 Yes, this test fails because it tries to delete an object in teardown that was already deleted. I haven't changed the logic behind teardown.

@petr-balogh Martin told me that you are planning to change how cleaning objects in teardown works. Is it true or should I change something here so the test won't fail when the fixture object is deleted in test?

Reply:
This is still under consideration but as you can see this changing logic because this class didn't use fixture in @pytest.mark.usefixtures decorator for deletion of pod (because test itself deleting the object) but now in the rbd_pod fixture you take care about deleting while origin fixture create_rbd_pod didn't have such finalizer.

I am proposing that in all the object we have where we have this delete function we can add flag (class atribute like self.deleted = True).

So when we will have finalizer we can check if the object was already deleted by this flag and if so the finalizer will skip its deletion.   WDYT?

Opened the PR for this flag here: #422 

So the logic needs to be changed here for sure.